### PR TITLE
Add flag for custom endpoint

### PR DESCRIPTION
**Pull request by hdhog**
This change provide sync for custom s3 storage.

Add Options
 - Endpoint
 - PathStyle

### CODE REVIEW

**Review by appleboy — APPROVED** [review]

File: main.go [hunk 1]
```diff
@@ -19,6 +19,11 @@ func main() {
 	app.Version = fmt.Sprintf("1.1.0+%s", build)
 	app.Flags = []cli.Flag{
 		cli.StringFlag{
+			Name:   "endpoint",
+			Usage:  "endpoint for the s3 connection",
+			EnvVar: "PLUGIN_ENDPOINT,S3_ENDPOINT",
```
Comment (tboerger) [inline main.go]: Maybe also add S3_SYNC_ENDPOINT to the middle

File: aws.go [hunk 2]
```diff
@@ -28,8 +29,11 @@ type AWS struct {
 
 func NewAWS(p *Plugin) AWS {
 	sess := session.New(&aws.Config{
-		Credentials: credentials.NewStaticCredentials(p.Key, p.Secret, ""),
-		Region:      aws.String(p.Region),
+		Endpoint:         &p.Endpoint,
+		DisableSSL:       aws.Bool(strings.HasPrefix(p.Endpoint, "http://")),
```
Comment (tboerger) [inline aws.go]: This should only be set if a custom endpoint have been defined

**Review by tboerger — CHANGES_REQUESTED** [review]

**Comment by hdhog** [comment]
@tboerger i fixed it

File: aws.go [hunk 3]
```diff
@@ -27,10 +28,19 @@ type AWS struct {
...
-		Credentials: credentials.NewStaticCredentials(p.Key, p.Secret, ""),
-		Region:      aws.String(p.Region),
-	})
+	sessCfg := &aws.Config{
+		Credentials:      credentials.NewStaticCredentials(p.Key, p.Secret, ""),
+		S3ForcePathStyle: aws.Bool(p.PathStyle),
+		Region:           aws.String(p.Region),
+	}
+
+	fmt.Println(p.Endpoint)
```
Comment (tboerger) [inline aws.go]: Forgot to drop this debugging output?

Comment (hdhog) [reply aws.go]: yes, fixed

**Comment by hdhog** [comment]
@tboerger, need some more changes?

**Review by tboerger — APPROVED** [review]

**Pull request merged**